internal/command/action_test.go

```diff
@@ -120,6 +128,141 @@ func TestRunAction(t *testing.T) {
 
 		So(string(content), ShouldEqual, expectedValue)
 	})
+
+	Convey("Docker options correctly injected for top-level command", t, func() {
+		RunDockerArgsTestCase(t, func (dockerDaemonSocket string) []string {
+			return []string{
+				"docker",
+				"-H", dockerDaemonSocket,
+				"run",
+				"--rm", "-d", "@SUMMONDOCKERARGS",
+				"alpine",
+			}
+		})
+	})
+
+	Convey("Docker options correctly injected for nested command", t, func() {
+		RunDockerArgsTestCase(t, func (dockerDaemonSocket string) []string {
+			return []string{
+				"sh",
+				"-c",
+				"docker -H "+dockerDaemonSocket+" run --rm -d @SUMMONDOCKERARGS alpine",
+			}
+		})
+	})
+}
+
+func RunDockerArgsTestCase(
+	t *testing.T,
+	dockerCommandGen func(dockerDaemonSocket string,
+) []string) {
+	// This is a test case for @SUMMONDOCKERARGS. It exercises Docker CLI pointed to a mock
+	// server. It asserts on the request payload received on the container creation
+	// endpoint, the volume mounts and environment variables injected by summon are
+	// expected to be present.
+
+	expected := map[string]string{
+		"A": "A's multiple line\nvalue",
+		"B": "B_value",
+		"C": "C_value",
+		"D": "D_value",
+	}
+	const inlineSecretsYml = `
+A: |-
+ A's multiple line
+ value
+B: !var B_value
+C: !file C_value
+D: !var:file D_value
+`
+
+	volumeBinds := map[string]struct{
+		ContainerPath string
+		FileContents string
+	}{}
+	envvars := map[string]string{}
+
+	// Mock server for handling API calls by `docker run`
```

Comment:
This is some slick docker daemon mocking. I haven't used http mocks in a while, this is a good refresher.